glib/src/param_spec.rs

```diff
@@ -384,6 +384,97 @@ macro_rules! define_param_spec_numeric {
     };
 }
 
+macro_rules! define_builder {
+    (@constructors $rust_type:ident, $builder_type:ident $(($($req_ident:ident: $req_ty:ty,)*))?) => {
+        impl<'a> $builder_type<'a> {
+            pub fn new(name: &'a str, $($($req_ident: $req_ty)*)?) -> Self {
+                Self {
+                    name,
+                    $($($req_ident: Some($req_ident),)*)?
+                    ..Default::default()
+                }
+            }
+        }
+
+        impl $rust_type {
+            pub fn builder<'a>(name: &'a str, $($($req_ident: $req_ty),*)?) -> $builder_type<'a> {
+                $builder_type::new(name, $($($req_ident),*)?)
+            }
+        }
+    };
+    (
+        $rust_type:ident, $builder_type:ident {
+            $($field_id:ident: $field_ty:ty $(= $field_expr:expr)?,)*
+        }
+        $(requires $required_tt:tt)?
+    ) => {
+        #[derive(Default)]
+        pub struct $builder_type<'a> {
+            name: &'a str,
+            nick: Option<&'a str>,
+            blurb: Option<&'a str>,
+            flags: Option<crate::ParamFlags>, // `ParamFlags` don't implement `Default`, so I wrap them in `Option`
```

Comment:
Should it implement Default as READWRITE?

Reply:
I don't like that. I feel bitflags should have 0, or another "base" state, as the default value. That makes working with bitflags easier.

In this context having `READWRITE` as default makes sense, in other context it doesn't.

Reply:
We could manually implement `Default` like this for `ParamFlags`. I think that would make sense

We currently don't derive/impl `Default` for flags so there's no problem here and this can be considered for other flags on a case-by-base basis.

Reply:
@ranfdev Can you put this into a separate commit, and then squash all the other commits into one?